job_bundles/Using_Files_In_Jobs.md

```diff
@@ -0,0 +1,1154 @@
+2024-07-17 01:26:38,174 INFO Downloaded 186.0 B / 186.0 B of 2 files (Transfer rate: 733.0 B/s)
+2024-07-17 01:26:38,176 INFO Summary Statistics for file downloads:
+Processed 2 files totaling 186.0 B.
+Skipped re-processing 0 files totaling 0.0 B.
+Total processing time of 0.09752 seconds at 1.91 KB/s.
+```
+
+Then, after that you will see the output from `script.sh` that was run by the job that shows:
+
+1. The input files that were uploaded when the job was submitted are located under a directory whose
+   name begins with "assetroot" within the session's temporary directory;
+2. That those input files' paths have been relocated to be relative to that "assetroot" directory
+   instead of relative to the root path for the job's input manifest (`"/"`); and
+3. The path mapping rules file contains an additional rule that remaps `"/"` to the absolute path
+   of the "assetroot" directory.
+
+For example:
+
+```
+2024-07-17 01:26:38,264 INFO Output:
+2024-07-17 01:26:38,267 INFO Session working directory is: /sessions/session-b041c12fb5ba4dc9aa978bf71138ac34qnysdvxy
+2024-07-17 01:26:38,267 INFO 
+2024-07-17 01:26:38,267 INFO Contents:
+2024-07-17 01:26:38,269 INFO ./tmp_xdhbsdo.sh
+2024-07-17 01:26:38,269 INFO ./tmpdi00052b.json
+2024-07-17 01:26:38,269 INFO ./assetroot-3751af5bd81011e6d21d/shared/projects/project2/file.txt
+2024-07-17 01:26:38,269 INFO ./assetroot-3751af5bd81011e6d21d/home/cloudshell-user/job_attachments_devguide/script.sh
+2024-07-17 01:26:38,269 INFO 
+2024-07-17 01:26:38,270 INFO Path mapping rules file: /sessions/session-b041c12fb5ba4dc9aa978bf71138ac34qnysdvxy/tmpdi00052b.json
+2024-07-17 01:26:38,282 INFO {
+2024-07-17 01:26:38,282 INFO   "version": "pathmapping-1.0",
+2024-07-17 01:26:38,282 INFO   "path_mapping_rules": [
+2024-07-17 01:26:38,282 INFO     {
+2024-07-17 01:26:38,282 INFO       "source_path_format": "POSIX",
+2024-07-17 01:26:38,282 INFO       "source_path": "/shared/projects/project1",
+2024-07-17 01:26:38,283 INFO       "destination_path": "/mnt/projects/project1"
+2024-07-17 01:26:38,283 INFO     },
+2024-07-17 01:26:38,283 INFO     {
+2024-07-17 01:26:38,283 INFO       "source_path_format": "POSIX",
+2024-07-17 01:26:38,283 INFO       "source_path": "/shared/common",
+2024-07-17 01:26:38,283 INFO       "destination_path": "/mnt/common"
+2024-07-17 01:26:38,283 INFO     },
+2024-07-17 01:26:38,283 INFO     {
+2024-07-17 01:26:38,283 INFO       "source_path_format": "POSIX",
+2024-07-17 01:26:38,283 INFO       "source_path": "/",
+2024-07-17 01:26:38,283 INFO       "destination_path": "/sessions/session-b041c12fb5ba4dc9aa978bf71138ac34qnysdvxy/assetroot-3751af5bd81011e6d21d"
+2024-07-17 01:26:38,283 INFO     }
+2024-07-17 01:26:38,283 INFO   ]
+2024-07-17 01:26:38,283 INFO }
+```
+
+Note that if the job that you submit has multiple manifests with different root paths, then there will be a different "assetroot"-named directory
```

Comment:
nit: This point is really important ! Different input directories here will have different asset roots and will be mapped accordingly. EG: A network system file path vs a local dir.

Can we move this note up?

This maybe an advanced example to add later with multiple asset roots.

Reply:
Good thinking. I think that this is a good place for the note, flow-wise, but I'll add some visual emphasis to draw attention to it.